lib/AsyncStorage.js

```diff
@@ -103,7 +103,19 @@ const AsyncStorage = {
         const errs = convertErrors(errors);
         callback && callback(errs && errs[0]);
         if (errs) {
-          reject(errs[0]);
+          if (Platform.OS !== 'ios') {
+            reject(errs[0]);
+          } else if (
+            errs[0] &&
+            String(errs[0]).match('Entries must be arrays of the form') &&
```

Comment:
Turns out, the error `Entries must be arrays of the form` from native was shown when somehow wrongly parsed data was given. Since I upgraded the RN, it's no longer showing up.

We can revert those changes back to checking if value is not type of string, then displaying our warning.

Reply:
@Krizzu I've reverted it to the previous version.